atlp-rwanda/phoenix-bn-backend
# User should be able to reset password

### What does this PR do?
> This PR helps users reset their password when they forgot it
### Description of Task to be completed
>-Check if the user with the email exists
>-Handle errors if the user doesn't exists
>-Send a reset link to the user if he/she exists
>-Create a reset token
>-Use the token to help user change his/her password
>-Handle errors if the token expired
>-Write api tests
### How should this be manually tested?
>-Clone the repos
>-Switch to the `ft-resetPassword` branch
>-Add the env variables in `.env` file
>-Your `.env` file should look like this : 
>     - `DB_PORT=`  
>     - `DB_HOST=`  
>     - `DB_USER=`
>     - `DB_PASSWORD=`
>     - `DB_NAME=`  
>     - `DB_DIALECT=`  
>     - `DATABASE_URL=`
>     - `JWTKEY=`
>     - `API_VERSION=v1`
>     - `EMAIL=nomadbarefoot2@gmail.com`
>     - `PASSWORD=nistnvpuhqalrrej`
>     - `HOST=http://localhost:5000`
>     - `PORT=5000`
> run `npm install` to install the required package dependances
> run `npm start` to run the application
### Any background context you want to provide?
> When the user enter his email, he/she will be able to receive a reset link in his email  
>  Once he/she clicks that link, he/she will be able to change his password
### What are the relevant Trello story?
> Ft(resetPassword >> https://trello.com/c/MxF6s3ol/22-reset-password
### Screenshots 
>When the user doesn't exist
![1](https://user-images.githubusercontent.com/49835290/96933815-022afc00-1476-11eb-9798-77802cb5cd96.PNG)
>When the user with the email exists
![2](https://user-images.githubusercontent.com/49835290/96933960-45856a80-1476-11eb-9d54-94a169bf4b31.PNG)
>When the token expired
![3](https://user-images.githubusercontent.com/49835290/96934202-b036a600-1476-11eb-8e22-a9220c2b61ce.PNG)
>Use the generated token in URL to change the password
![4](https://user-images.githubusercontent.com/49835290/96934420-03a8f400-1477-11eb-8702-aa21608c1e1f.PNG)
